src/core/protocol.ts

```diff
@@ -145,3 +40,101 @@ export class Protocol {
     return request;
   }
 }
+
+export class RedisProtocolError extends Error {
+  public static fromMessage(message: string) {
+    const space = message.indexOf(" ");
+    return new RedisProtocolError(message.slice(0, space), message.slice(space + 1));
+  }
+  constructor(name: string, message: string) {
+    super();
+    this.name = name;
+    this.message = message;
+  }
+}
+
+class ProtocolParser {
+  public static parse(raw: string): InterfaceParser {
+    let masterRegex = [
+      "\\+(?<simple>.+?)",
+      "\\-(?<error>.+?)",
+      "\\:(?<int>-?\\d+)",
+      "(?<blobString>\\$blobRef_\\d+)", // Note: this matches our replaced Blob-Ref
+      "\\*(?<array_n>\\d+)(?<array>)",
+      "(\\$|\\*)(?<null_string>-1)",
+    ].join("|");
+
+    // Preprocess blobs since they can include <CR><LF>
+    const blobs = ProtocolParser.extractBlobs(raw, "$");
+
```

Comment:
In working on support for RESP3, I realized I wasn't handling Bulk Strings correctly. `extractBlobs` now pulls out Bulk Strings ("Blob Strings" in RESP3) and replaces them with a simple key which is then expanded during aggregation to the full blob.  This allows Bulk Strings to contain `\r\n`, whereas before the Regex wouldn't have handled that correctly.